src/Emulator/Extensions/Mocks/HPSHostController.cs

```diff
@@ -171,6 +174,40 @@ private byte[] GetBytesFromSlave(int count, TimeInterval timeInterval)
             return result;
         }
 
+        // This method is for getting OA1EN, 
+        // returns true when OA1EN is enabled.
+        private bool OA1ENEnabled(){
+            return currentDoubleSlave.ReadDoubleWord(0x08) >> 15 != 0x00;
+        }
+
+        // This method is for getting RXEN,
+        // returns true when the buffer is empty.
+        private bool RXNECleared(){
+            return (currentDoubleSlave.ReadDoubleWord(0x18) & (1 << 2)) == 0x00;
+        }
+        
+        private delegate bool RegisterBitAccessDelegate();
```

Comment:
Instead of introducing a new delegate type you can use a built-in `Func<bool>` type.